Source/NSLayoutConstraint.m

```diff
@@ -0,0 +1,601 @@
+    }  
+  else if ([@"Height" isEqualToString: str])
+    {
+      a = NSLayoutAttributeHeight;
+    }
+  else if ([@"CenterX" isEqualToString: str])
+    {
+      a = NSLayoutAttributeCenterX;
+    }  
+  else if ([@"CenterY" isEqualToString: str])
+    {
+      a = NSLayoutAttributeCenterY;
+    }
+  else if ([@"Baseline" isEqualToString: str])
+    {
+      a = NSLayoutAttributeBaseline;
+    }  
+  else if ([@"FirstBaseline" isEqualToString: str])
+    {
+      a = NSLayoutAttributeFirstBaseline;
+    }
+  else if ([@"NotAnAttribute" isEqualToString: str])
+    {
+      a = NSLayoutAttributeNotAnAttribute;
+    }
+
+  return a;
+}
+
++ (NSString *) _relationToString: (NSLayoutRelation)rel
+{
+  NSString *relation = nil;
+
+  switch (rel)
+    {
+    case NSLayoutRelationLessThanOrEqual:
+      relation = @"<=";
+      break;
+    case NSLayoutRelationEqual:
+      relation = @"=";
+      break;
+    case NSLayoutRelationGreaterThanOrEqual:
+      relation = @">=";
+      break;
+    default:
+      break;
+    }
+
+  return relation;
+}
+
++ (NSLayoutRelation) _stringToRelation: (NSString *)str
+{
+  NSLayoutRelation r = 0;
+
+  if ([@"<=" isEqualToString: str])
+    {
+      r = NSLayoutRelationLessThanOrEqual;
+    }
+  else if ([@"=" isEqualToString: str])
+    {
+      r = NSLayoutRelationEqual;
+    }
+  else if ([@">=" isEqualToString: str])
+    {
+      r = NSLayoutRelationGreaterThanOrEqual;
+    }
+  
+  return r;
+}
+
++ (void) _activateConstraint: (NSLayoutConstraint *)constraint
+{
+  [activeConstraints addObject: constraint];
+  activeConstraints = [[activeConstraints sortedArrayUsingSelector: @selector(compare:)] mutableCopy];
+}
+
++ (void) _removeConstraint: (NSLayoutConstraint *)constraint
+{
+  [activeConstraints removeObject: constraint];
+}
+
++ (NSArray *) constraintsWithVisualFormat: (NSString *)fmt 
+                                  options: (NSLayoutFormatOptions)opt 
+                                  metrics: (NSDictionary *)metrics 
+                                    views: (NSDictionary *)views
+{
+  NSMutableArray *array = [NSMutableArray arrayWithCapacity: 10];
+  return array;
+}
+
+- (instancetype) initWithItem: (id)firstItem 
+                    attribute: (NSLayoutAttribute)firstAttribute 
+                    relatedBy: (NSLayoutRelation)relation 
+                       toItem: (id)secondItem
+                    attribute: (NSLayoutAttribute)secondAttribute 
+                   multiplier: (CGFloat)multiplier
+                     constant: (CGFloat)constant
+                     priority: (CGFloat)priority;
+{
+  self = [super init];
+  if (self != nil)
+    { 
+      _firstItem = firstItem;
+      _secondItem = secondItem;
+      _firstAttribute = firstAttribute;
+      _secondAttribute = secondAttribute;
+      _relation = relation;
+      _multiplier = multiplier;
+      _constant = constant;
+      _priority = priority;
+      
+      [NSLayoutConstraint _activateConstraint: self];
```

Comment:
I don't think that constraints are active by default. Looking at the example code from Apple there constraints get activated explicitly after being created in the code.

Reply:
Do you think they are activated by default when unarchived?  It seems to me that they are.

Reply:
Yes, this is the most likely case.